### ISSUE TITLE
Find Local Peaks failing on Pro 2.1 

### ISSUE THREAD

**Issue by csmoore** [Esri/military-tools-geoprocessing-toolbox#260]
Find Local Peaks (~~and Line Of Sight~~ update: no longer failing) is failing on Pro 2.1

Tested on Pro 2.1.10044

![image](https://user-images.githubusercontent.com/3090809/32671906-22e6fba2-c617-11e7-8b8b-99529699b9ea.png)

```
ERROR: 2017-11-10 09:16:45,744 ERRORS ==================================================
ERROR: 2017-11-10 09:16:45,746 test_findLocalPeaks (visibility_tests.VisibilityUtilitiesTestCase.VisibilityUtilitiesTestCase)
ERROR: 2017-11-10 09:16:45,747 Traceback (most recent call last):
ERROR: 2017-11-10 09:16:45,748   File "C:\Users\chri0020\Documents\Github\military-tools-geoprocessing-toolbox-csm\utils\test\visibility_tests\VisibilityUtilitiesTestCase.py", line 360, in test_findLocalPeaks
ERROR: 2017-11-10 09:16:45,748     self.assertTrue(arcpy.Exists(resultPoints), "Output features do not exist or were not created")
ERROR: 2017-11-10 09:16:45,749   File "C:\Program Files\ArcGIS\Pro\Resources\ArcPy\arcpy\__init__.py", line 836, in Exists
ERROR: 2017-11-10 09:16:45,751     return gp.exists(dataset)
ERROR: 2017-11-10 09:16:45,755   File "C:\Program Files\ArcGIS\Pro\Resources\ArcPy\arcpy\geoprocessing\_base.py", line 303, in exists
ERROR: 2017-11-10 09:16:45,756     self._gp.Exists(*gp_fixargs(args, True)))
ERROR: 2017-11-10 09:16:45,756 AttributeError: Object: Error in parsing arguments for Exists
ERROR: 2017-11-10 09:16:45,757 test_find_local_peaks (visibility_tests.FindLocalPeaksTestCase.FindLocalPeaksTestCase)
ERROR: 2017-11-10 09:16:45,758 Traceback (most recent call last):
ERROR: 2017-11-10 09:16:45,758   File "C:\Users\chri0020\Documents\Github\military-tools-geoprocessing-toolbox-csm\utils\test\visibility_tests\FindLocalPeaksTestCase.py", line 92, in test_find_local_peaks
ERROR: 2017-11-10 09:16:45,759     arcpy.FindLocalPeaks_mt(self.inputArea, 10, self.inputSurface, self.outputPoints)
ERROR: 2017-11-10 09:16:45,760   File "C:\Users\chri0020\Documents\Github\military-tools-geoprocessing-toolbox-csm\toolboxes\Military_Tools_pro.tbx", line 644, in FindLocalPeaks
ERROR: 2017-11-10 09:16:45,760     8M/Aev8Axf8AGM3xS+Jc39oajd7TZWbRlIY4lJMaRxkny4U3HYmSSSzsWZmZvoSkVQqhVGAOABS1
ERROR: 2017-11-10 09:16:45,761   File "C:\Users\chri0020\Documents\Github\military-tools-geoprocessing-toolbox-csm\toolboxes\Military_Tools_pro.tbx", line 641, in FindLocalPeaks
ERROR: 2017-11-10 09:16:45,762     UUAFFFFABRRRQAV5t+0B8ctK/Z/+H0niTUbSfUrmadbHTtOgypurpkd0RpMERJtjdmdgcKh2h3Ko
ERROR: 2017-11-10 09:16:45,762   File "C:\Program Files\ArcGIS\Pro\Resources\ArcPy\arcpy\geoprocessing\_base.py", line 506, in <lambda>
ERROR: 2017-11-10 09:16:45,763     return lambda *args: val(*gp_fixargs(args, True))
ERROR: 2017-11-10 09:16:45,764 arcgisscripting.ExecuteError: Start Time: Friday, November 10, 2017 9:13:13 AM
ERROR: 2017-11-10 09:16:45,768 Failed to execute. Parameters are not valid.
ERROR: 2017-11-10 09:16:45,768 ERROR 000824: The tool is not licensed.
ERROR: 2017-11-10 09:16:45,769 Failed to execute (FeatureToPoint).
ERROR: 2017-11-10 09:16:45,770 Failed at Friday, November 10, 2017 9:13:13 AM (Elapsed Time: 0.00 seconds)
ERROR: 2017-11-10 09:16:45,770 Failed to execute (FindLocalPeaks).
ERROR: 2017-11-10 09:16:45,771 test_linear_line_of_sight (visibility_tests.LinearLineOfSightTestCase.LinearLineOfSightTestCase)
ERROR: 2017-11-10 09:16:45,772 Traceback (most recent call last):
ERROR: 2017-11-10 09:16:45,772   File "C:\Users\chri0020\Documents\Github\military-tools-geoprocessing-toolbox-csm\utils\test\visibility_tests\LinearLineOfSightTestCase.py", line 113, in test_linear_line_of_sight
ERROR: 2017-11-10 09:16:45,773     self.outputTargets)
ERROR: 2017-11-10 09:16:45,773   File "C:\Users\chri0020\Documents\Github\military-tools-geoprocessing-toolbox-csm\toolboxes\Military_Tools_pro.tbx", line 722, in LinearLineOfSight
ERROR: 2017-11-10 09:16:45,775     7MpNSV0FFFFSMKKKKACs3V/DWla8m3UdOt7wdMyxgkfQ9RWlRTUnF3Qmk9Geaan8BdDlm+0aVc3m
ERROR: 2017-11-10 09:16:45,776   File "C:\Users\chri0020\Documents\Github\military-tools-geoprocessing-toolbox-csm\toolboxes\Military_Tools_pro.tbx", line 719, in LinearLineOfSight
ERROR: 2017-11-10 09:16:45,781     3g/miH8zntXTWdnBp9rHb20SQQRjakcYwqj2FS0USlzaLRBGNtXuFFFFZlhRRRQAUUUUARXlpDf2
ERROR: 2017-11-10 09:16:45,782   File "C:\Program Files\ArcGIS\Pro\Resources\ArcPy\arcpy\geoprocessing\_base.py", line 506, in <lambda>
ERROR: 2017-11-10 09:16:45,783     return lambda *args: val(*gp_fixargs(args, True))
ERROR: 2017-11-10 09:16:45,784 arcgisscripting.ExecuteError: Start Time: Friday, November 10, 2017 9:15:00 AM
ERROR: 2017-11-10 09:16:45,785 Failed to execute. Parameters are not valid.
ERROR: 2017-11-10 09:16:45,786 ERROR 000824: The tool is not licensed.
ERROR: 2017-11-10 09:16:45,787 Failed to execute (FeatureVerticesToPoints).
ERROR: 2017-11-10 09:16:45,788 Failed at Friday, November 10, 2017 9:15:00 AM (Elapsed Time: 0.00 seconds)
ERROR: 2017-11-10 09:16:45,792 Failed to execute (LinearLineOfSight).
INFO: 2017-11-10 09:16:45,793 END OF TEST =========================================
```

**Comment by lfunkhouser** [comment]
@csmoore can you dig into this more and give an explanation for the failure/errors?

**Comment by csmoore** [comment]
I do see the problem here at Pro 2.1, the raster outputs of [the Spatial Analyst calls](https://github.com/Esri/military-tools-geoprocessing-toolbox/blob/dev/toolboxes/scripts/VisibilityUtilities.py#L1007) no longer have the attribute tables built - so the next step ("CountRows") fails..

Since every other version of the product until Pro 2.1 (ArcMap, < Pro 2.1) did do this - this may be an issue we should report - and not intended behavior.

If we can't fix this in Pro 2.1, we will have to call "Build Attribute Table" GP Tool separately. Ex:
Add: if Pro version >= 2.1 gp.BuildRasterAttributeTable calls to any SA-dependent tools that require an attribute table.

FYI: @BobBooth @mfunk @adgiles - just in case I am missing anything

**Comment by csmoore** [comment]
This is still an issue as of 2.1.10195

**Comment by mfunk** [comment]
Just a cursory glance at this issue now...
So the ```ERROR 000824: The tool is not licensed.``` was not an error in the recent builds?
W.r.t the raster attribute tables, if that is the problem you should not have any issue adding BuildRasterAttribute table as it was supported back at 10.3 and probably earlier. Also the "OVERWRITE" option should be in all caps for backwards support for Desktop. The framework used to be really picky about stuff like this. Not sure if Pro will accept any form of it or not. Just a thought.

**Comment by topowright** [comment]
In trying to understand this issue further I have tested it on the released version of ArcGIS Pro 2.1 and it does not fail; however, this tool failed on previous versions of ArcGIS Pro 2.1 for me as well. 

version: 2.1.0
Data:  ElevationUTM_Zone10 (Located in the GDB test data)
output: used the scratch workspace default

I am moving this to done and marking it invalid being that it is no longer a bug in the released version.
